### ISSUE TITLE
Will support squared-hinged like in sk-learn?

### ISSUE THREAD

**Issue by b04505009** [Xtra-Computing/thundersvm#100]
As title,
Will thundersvm support squared-hinged like in sk-learn?
Thanks!

**Comment by zeyiwen** [comment]
Thanks for the reminding us! 

We plan to support this loss in the library. However, we haven't got many requests for this functionality. Our current primary task is supporting ```structured output```. 

Perhaps, we can work on this loss next. Could you share some insight or applications why you think squared hinge loss is more appealing than hinge loss? This would help us prioritize the tasks.

**Comment by b04505009** [comment]
[LinearSVC](http://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html) in sk-learn use squared-hinged as default.

[Here](https://cvstuff.wordpress.com/2014/11/29/latex-l_1-versus-latex-l_2-loss-a-svm-example/) said that squared-hinge is more likely a regressor while that of hinge is more likely a classifier.
However, in my case of classification, squared-hinge is acting a little better.

Maybe the best result for me is to run the LinearSVC on GPU. LinearSVC use different library (LibLinear) from SVC's library (LibSVM), which is optimized for linear case.
Though LinearSVC is much faster than SVM, it still becomes slow when choosing larger C.
Moving this library on GPU may be helpful.

**Comment by zeyiwen** [comment]
Thank you for the feedback! The blog you shared is not totally correct. The author was confused between ```squared loss``` and ```squared hinge loss```. 

You have mentioned a very good point. We should support LinearSVC which will be a GPU version for LIBLINEAR. We will definitely support this, but it will take some time. Please stay tuned!
